website/blog/corba.md

```diff
@@ -0,0 +1,138 @@
+entities called [components][]. A component implements a set of methods, and
+components interact by calling these methods. When a Service Weaver application
+is deployed, some components are co-located in the same process, and others are
+run in separate processes, potentially on separate machines. As a result, some
+method calls are performed locally, and some are performed remotely.
+
+This paradigm might sound familiar to you. It's similar to the distributed
+object model used by CORBA, DCOM, Java RMI, and countless other frameworks that
+have come and gone. It is [widely believed][why_corba_failed] that these
+frameworks failed because they attempt to make remote method calls **look** and
+**act** the same as local method calls. This flies in the face of the [fallacies
+of distributed computing][fallacies], right? There are [whole
+papers][a_note_on_dc] explaining that frameworks trying to unify local and
+remote objects are "doomed to failure". Does this mean that Service Weaver is
+doomed to fail just like CORBA?
+
+No.
+
+In this blog post, we explain that remote method calls in Service Weaver
+do **look** like local calls, but that's actually okay. They don't, however,
+**act** like local method calls, which is where a framework can get in trouble.
+We'll also explain that CORBA didn't actually fail for any these reasons and
+why Service Weaver is not "doomed" to the same fate.
+
+## Looking Local
+
+- When we say a remote method call **looks** like a local method call, we mean
+  they **syntactically** look the same. We discuss the semantics of the calls,
+  or the way they act, in the next section.
+- Typically, remote method calls are syntactically pronounced. To execute an RPC
+  with gRPC, for example, you first have to construct [an RPC
+  client](https://grpc.io/docs/languages/go/basics/#creating-a-stub), making it
+  clear that a call is about to happen over the network. Then, when you issue
+  the RPC, you have to [pass protobufs as
+  arguments](https://grpc.io/docs/languages/go/basics/#simple-rpc-1), again
+  making it clear that the arguments are being serialized and sent over the
+  network.
+- Some frameworks choose to add additional syntax to remote calls even when it
+  is otherwise not needed. Ray, for example, requires that a function `f` be
+  called as
+  [`f.remote`](https://docs.ray.io/en/latest/ray-core/walkthrough.html#running-a-task)
+  when invoked remotely.
+- However, placing a syntactic burden on developers to make remote method calls
+  is unnecessary. Even [the paper][a_note_on_dc] which describes why unifying
+  local and remote objects is doomed to fail does "not argue against pleasant
+  programming interfaces."
+- Some people argue that a syntactic marker for remote method calls is needed to
+  see that a remote call is happening. However, any syntactic marker is
+  inadequate, as a remote method call can be easily wrapped inside a function,
```

Comment:
Maybe change "can be easily wrapped" to something like "is usually wrapped".

And I wonder if we can find some common piece of OSS code that we can point at as an example where a call looks local bit it wraps inside it something that is remote.

Also we could point out that DB calls are sometimes local and sometimes remote (sqlite backend for Go vs. say Postgres).